core/controlplane/config/templates/cloud-config-controller

```diff
@@ -4803,6 +4813,16 @@ write_files:
   - path: /etc/kubernetes/ssl/service-account-key.pem{{if .AssetsEncryptionEnabled}}.enc{{end}}
     encoding: gzip+base64
     content: {{.AssetsConfig.ServiceAccountKey}}
+
+  {{ if .Addons.MetricsServer.Enabled -}}
```

Comment:
Api agreggation layer is used nowadays for several things not only for metrics-server i think is a good idea to generate it and configure it even if you disable metrics-server